### PR TITLE
Disable incremental marking on Node > 8.4.0

### PR DESCRIPTION
**Pull request by fs-eire**
Change #125 force failure on Node > 8.4.0 because of compability issue #96.

Now we found a workaround from https://github.com/nodejs/node/issues/16658.

Napa.js will use this workaround: set V8 flag `--noincremental-marking` on Node >= 8.4.0, instead of throwing an exception.

### CODE REVIEW

File: lib/binding.js [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
-function failCheck(messages) {
+function warn(messages) {
     messages.forEach(function(message) {
         console.log('\x1b[1m\x1b[33m', message, '\x1b[0m');
     });
-    throw new Error(messages.join(' '));
+}
+
+function fail(messages) {
+    warn(messages);
```
Comment (helloshuangzi) [inline lib/binding.js]: use red color for failing message to differentiate it with warning messages which is yellow?

Comment (fs-eire) [reply lib/binding.js]: What do you think of using [npmlog](https://www.npmjs.com/package/npmlog)? It is already the dependency of Napa.js now.

Comment (helloshuangzi) [reply lib/binding.js]: good idea!

File: lib/binding.js [hunk 2]
```diff
@@ -27,4 +22,4 @@ if (typeof __in_napa === 'undefined') {
     module.exports = require('../bin/napa-binding');
 } else {
     module.exports = process.binding('napa-binding');
-}
+}
```
Comment (helloshuangzi) [inline lib/binding.js]: a new line as EOF please?

**Comment by leolorenzoluis** [comment]
@fs-eire Does this mean NodeJS from > 8.4.0 up until 10.0.x will have this compatibility issue? What are the side effects of having --noincremental-marking disabling V8 incremental marking in a production app?

**Comment by fs-eire** [comment]
For the first question, the answer is YES.
For the second question, 'incremental marking' is an improvement to reduce the pause time in GC. Disabling this feature may cause a higher pause time in a GC operation, which may potentially impact the performance. The actual impact, however, highly depends on the machine spec, throughput and the program behavior. You can use some profiling tools to figure out the detailed value.

A good news is, Node v10.1.0 exposes an API which potentially can help. If it works, we can re-enable incremental marking in Node > 10.1.0.

**Pull request merged**